### ISSUE TITLE
|uninstall and |suspend function incorrectly (kiln)

### ISSUE THREAD

**Issue by hanfel-dovned** [urbit/urbit#6720]
**Describe the Bug**
There's some general weirdness with the relationship between kiln's `zyn` state and the `%zest` notes that kiln passes to Clay. This manifests in the following issues:
- `|uninstall` does not prevent the app publisher from pushing changes to the subscriber's desk (at least according to kiln output).
- `|suspend`ing an app *does* prevent the app publisher from pushing changes to the subscriber's desk, in such a way that `|revive`ing doesn't restore, preventing future app updates

**Reproducing the `|uninstall` bug:**
1. Create two fake ships, ~zod and ~dev
2. Install and publish a desk on ~zod via mount/commit/install our/treaty publish
3. Install that desk on ~dev: `|install ~zod %hello`
4. Uninstall the desk on ~dev: `|uninstall %hello`
5. Commit changes to the desk on ~zod
6. Notice in ~dev's terminal that kiln still produces output suggesting that it downloaded the update for this desk from the publisher. (See [this line](https://github.com/urbit/urbit/blob/10ef0c23410bf275718d42d848995ecbcb690dc5/pkg/arvo/lib/hood/kiln.hoon#L1131) in kiln.) However, if you mount the desk and check your filesystem, you'll see that these changes didn't actually get committed.

**Reproducing the `|revive` bug:**
1. Create two fake ships, ~zod and ~dev
2. Install and publish a desk on ~zod via mount/commit/install our/treaty publish
3. Install that desk on ~dev: `|install ~zod %hello`
4. Suspend the desk on ~dev: `|suspend %hello`
5. Revive the desk on ~dev: `|revive %hello`
6. Commit changes to the desk on ~zod
7. Notice in ~dev's terminal that there is no kiln output and that the desk was not updated.

**Expected Behavior**
- As per the [docs](https://operators.urbit.org/manual/os/dojo-tools#uninstall), `|uninstall` should ignore updates from the desk publisher. I would expect that uninstalling an app would terminate my relationship with its distributor and not allow them to put data on my ship anymore, even if that data is inert. It seems likely that this data isn't in fact getting put on the ship, but in that case kiln shouldn't be outputting that it has.
- `|suspend` should not unsubscribe permanently in such a way that `|revive` doesn't recover from. I don't expect to have to reinstall an app after turning it back on in order to get updates.

**System:**
 - Ubuntu 18.04.5
 - `vere 2.11`, `zuse 413k`
 - `%base` hash: `p8b1a`

**Additional context**
- It looks like the relevant parts of kiln to look at here are its poke handling arms, paying specific attention to when it deletes items from `zyn` and when it passes cards to Clay.
- `|pause` seems to be implicated in this situation as well. `|resume` doesn't function at all.

**Comment by MarcusMiguel** [comment]
@hanfel-dovned  I can't reproduce the `|revive` bug. The `suspended`/`revived` desk always receives updates. Can you please check if those steps are right?

[`|resume` **was removed** ](https://github.com/urbit/urbit/pull/6675#issue-1761653564). And `|pause` is more like 'unsync' than 'pause'.

**Comment by hanfel-dovned** [comment]
You're right, it looks like you need to `|uninstall` as well in order to stop the updates:

1. Create two fake ships, ~zod and ~dev
2 Install and publish a desk on ~zod via mount/commit/install our/treaty publish
3. Install that desk on ~dev: |install ~zod %hello
4. Uninstall the desk on ~dev: |uninstall %hello
5. Suspend the desk on ~dev: |suspend %hello
6. Revive the desk on ~dev: |revive %hello
7. Commit changes to the desk on ~zod
8. Notice in ~dev's terminal that there is no kiln output and that the desk was not updated.

This is in contrast to uninstalling it but not suspending/reviving, in which case you do still get updates.

Less of an issue than I initially thought, but I'd still expect the same thing to happen after suspending/reviving as prior to suspending/reviving, regardless of if the app is installed or uninstalled.

**Comment by MarcusMiguel** [comment]
The fix for the`|uninstall` bug will be out in the next kelvin. I'm closing this since i don't think that there is anything else to do here.
